main.tf

```diff
@@ -569,9 +569,17 @@ resource "aws_subnet" "elasticache" {
 resource "aws_elasticache_subnet_group" "elasticache" {
   count = var.create_vpc && length(var.elasticache_subnets) > 0 && var.create_elasticache_subnet_group ? 1 : 0
 
-  name        = var.name
+  name        = lower(coalesce(var.elasticache_subnet_group_name, var.name))
   description = "ElastiCache subnet group for ${var.name}"
   subnet_ids  = aws_subnet.elasticache.*.id
+
+  tags = merge(
```

Comment:
Make sure to update `versions.tf` accordingly. `tags` were added in v3.38.0 of Terraform AWS provider ([link](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#3380-april-30-2021))